server/player/player.go

```diff
@@ -1581,7 +1581,17 @@ func (p *Player) OpenBlockContainer(pos cube.Pos) {
 	if p.session() == session.Nop {
 		return
 	}
-	p.session().OpenBlockContainer(pos)
+	ctx := event.C()
+	p.handler().HandleContainerOpen(ctx, pos)
+
+	ctx.Continue(func() {
+		p.session().OpenBlockContainer(pos)
+	})
+}
+
+// CloseCurrentBlockContainer handles the closing of a block container by the player at the position passed.
+func (p *Player) CloseCurrentBlockContainer(pos cube.Pos) {
+	p.handler().HandleContainerClose(pos)
```

Comment:
This method does nothing by itself. Code in `session/player.go` should probably be moved here or this method name will be inaccurate.